RiotSharp/RiotApi.cs

```diff
@@ -301,73 +302,52 @@ public List<MasteryPage> GetMasteryPages(Region region, long summonerId)
         #endregion
 
         #region League
-        public Dictionary<long, List<League>> GetLeagues(Region region, List<long> summonerIds)
+        public List<League> GetLeagues(Region region, long summonerIds)
         {
-            var dict = new Dictionary<long, List<League>>();
-            foreach (var grp in MakeGroups(summonerIds, MaxNrLeagues))
-            {
                 var json = requester.CreateGetRequest(
                     string.Format(LeagueRootUrl, region.ToString()) +
-                        string.Format(LeagueBySummonerUrl, Util.BuildIdsString(grp)) + LeagueEntryUrl,
+                        string.Format(LeagueBySummonerUrl, summonerIds),
                     region);
-                var subDict = JsonConvert.DeserializeObject<Dictionary<long, List<League>>>(json);
-                foreach (var child in subDict)
-                {
-                    dict.Add(child.Key, child.Value);
-                }
-            }
-            return dict;
+                var list = JsonConvert.DeserializeObject<List<League>>(json);
+            return list;
         }
  
-        public async Task<Dictionary<long, List<League>>> GetLeaguesAsync(Region region, List<long> summonerIds)
+        public async Task<List<League>> GetLeaguesAsync(Region region, long summonerId)
         {
-            var tasks = MakeGroups(summonerIds, MaxNrLeagues).Select(
-                grp => requester.CreateGetRequestAsync(
+            var json = await requester.CreateGetRequestAsync(
                     string.Format(LeagueRootUrl, region.ToString()) +
```

Comment:
You've updated `LeagueRootUrl` to "/lol/league/v3", which contains no format placeholders, so I don't think `string.Format(LeagueRootUrl, region.ToString())` would work correctly.